cachi2/core/package_managers/yarn_classic/main.py

```diff
@@ -0,0 +1,16 @@
+from cachi2.core.models.output import EnvironmentVariable
+
+
+def _generate_build_environment_variables() -> list[EnvironmentVariable]:
+    """Generate environment variables that will be used for building the project.
+
+    These insure that yarnv1 will
+    - use the cachi2 cache
+    - not prune the cache
```

Comment:
It might be a good idea to explain somewhere (maybe the commit message?) why this is important.

Reply:
Maybe add something from the official docs as well:  https://classic.yarnpkg.com/en/docs/yarnrc